iree/tools/CMakeLists.txt

```diff
@@ -303,40 +330,22 @@ if(${IREE_BUILD_COMPILER})
       iree::vm::bytecode_module
       iree::vm::value
       ${IREE_HAL_DRIVER_MODULES}
+    HOSTONLY
   )
+endif(${IREE_BUILD_COMPILER})
 
-  iree_cc_library(
-    NAME
-      iree_translate_main
-    SRCS
-      "translate_main.cc"
-    DEPS
-      ::init_compiler_modules
-      ::init_iree_passes_and_dialects
-      ::init_mlir_passes_and_dialects
-      ::init_targets
-      ::init_translations
-      ::init_xla_dialects
-      LLVMSupport
-      MLIRIR
-      MLIRSCFTransforms
-      MLIRPass
-      MLIRSupport
-      MLIRTranslation
-      iree::compiler::Conversion::init_conversions
-      iree::compiler::Dialect::VM::Target::Bytecode
-      iree::compiler::Dialect::VM::Target::init_targets
-      iree::compiler::Translation::IREEVM
-    PUBLIC
-  )
-
+# If cross-compiling, we need to declare iree-translate under host configuration
+# unconditionally because we need to run it on host to generate VM modules
+# for tests.
+if(${IREE_BUILD_COMPILER} OR CMAKE_CROSSCOMPILING)
```

Comment:
But under host configuration IREE_BUILD_COMPILER is on for this reason. Wouldn't this turn it on for the target even if IREE_BUILD_COMPILER is off??

Reply:
Yep when compiling under host configuration `IREE_BUILD_COMPILER` is always set to on: these tool will be compiled. But when compiling for target configuration we still need to have these tools "defined" so we can find the host tool; otherwise under target configuration these tools will be missing. Here `CMAKE_CROSS_COMPILING` is for the latter.

(In the future we might want to enable compiling and running these tools under target configuration; then the target configurations' `IREE_BUILD_COMPILER` will also set to on.)